src/control/server/instance_storage_rpc.go

```diff
@@ -130,9 +132,18 @@ func (ei *EngineInstance) StorageFormatSCM(reformat bool) (mResult *ctlpb.ScmMou
 	}()
 
 	if ei.isStarted() {
-		scmErr = errors.Errorf("instance %d: can't format storage of running instance",
-			engineIdx)
-		return
+		if !reformat {
+			scmErr = errors.Errorf("instance %d: can't format storage of running instance",
+				engineIdx)
+			return
+		}
+
+		ei.log.Infof("forcibly stopping instance %d prior to reformat", ei.Index())
+		if scmErr = ei.Stop(unix.SIGKILL); scmErr != nil {
+			return
+		}
+
+		ei.requestStart(ctx)
```

Comment:
May be missing something here, but wouldn't this just start the engine right back up?

Reply:
Sort of... I think that we can revisit some of this stuff as part of Tom's refactoring. It's requesting that the instance control loop should take another turn. There is a bit of weird coupling between different parts of the code that I'm not happy about... There is backpressure from the runner which wants to write the final error from Wait() to a channel and that manifests in blocking over here where requestStart() is called. It took me quite a while to understand the relationship because it's not at all intuitive, but I wanted to keep changes to the bare minimum necessary to get this PR finished for 1.2.